service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -395,4 +397,52 @@ protected final CorsHandler createCorsHandler(final String allowedOrigin, final
                 .allowedHeader(HttpHeaders.IF_MATCH.toString())
                 .exposedHeader(HttpHeaders.ETAG.toString());
     }
+
+    /**
+     * Creates a response based on generic result.
+     * <p>
+     * The default behavior is as follows:
+     * <ol>
+     * <li>Set the status code on the response.</li>
+     * <li>If the status code represents an error condition (i.e. the code is &gt;= 400),
+     * then the JSON object passed in to the returned handler is written to the response body.</li>
+     * <li>Otherwise, the JSON object is written to the response body and the given custom handler is
+     * invoked (if not {@code null}).</li>
+     * </ol>
+     *
+     * @param ctx The routing context of the request.
+     * @param result The generic result of the operation.
+     * @param customHandler An (optional) handler for post processing the HTTP response, e.g. to set any additional HTTP
+     *                      headers. The handler <em>must not</em> write to response body. May be {@code null}.
+     */
+    protected final void createResponse(final RoutingContext ctx, final Result<?> result, final Handler<HttpServerResponse> customHandler) {
```

Comment:
*createResponse* to me sounds like an object would be created and returned. This is obviously not the case so what about renaming this to e.g. *writeResponse* ?